packages/flutter/lib/src/widgets/two_pane.dart

```diff
@@ -0,0 +1,482 @@
+  ///     screen into sub-screens.
+  ///   * If multiple are found in the same direction, the same rule applies.
+  ///   * If multiple are found in both directions, then the provided
+  ///     [direction] is used.
+  Axis _resolveDirection(Iterable<DisplayFeature> displayFeatures, Rect position) {
+    final Iterable<DisplayFeature> separatingFeatures = _separatingDisplayFeatures(displayFeatures, position);
+    bool verticalSubScreensExist = false;
+    bool horizontalSubScreensExist = false;
+    for (final DisplayFeature displayFeature in separatingFeatures){
+      if (_splitsHorizontally(displayFeature.bounds, position)) {
+        horizontalSubScreensExist = true;
+      }
+      if (_splitsVertically(displayFeature.bounds, position)) {
+        verticalSubScreensExist = true;
+      }
+    }
+    if (verticalSubScreensExist && !horizontalSubScreensExist) {
+      return Axis.vertical;
+    } else if (horizontalSubScreensExist && !verticalSubScreensExist) {
+      return Axis.horizontal;
+    } else {
+      return direction;
+    }
+  }
+
+  /// Retrieves the first [DisplayFeature] that splits the screen into separate
+  /// sub-screens.
+  ///
+  /// In case there are multiple display features, [textDirection] or
+  /// [verticalDirection] are used to determine the first one, according
+  /// [direction].
+  static DisplayFeature? _firstDisplayFeature(Iterable<DisplayFeature> displayFeatures,
+      Axis direction, TextDirection textDirection, VerticalDirection verticalDirection) {
+    if (displayFeatures.isEmpty) {
+      return null;
+    }
+    DisplayFeature result = displayFeatures.first;
+    for (final DisplayFeature displayFeature in displayFeatures){
+      switch (direction) {
+        case Axis.horizontal: {
+          switch (textDirection) {
+            case TextDirection.ltr: {
+              if (displayFeature.bounds.left < result.bounds.left) {
+                result = displayFeature;
+              }
+            }
+            break;
+            case TextDirection.rtl: {
+              if (displayFeature.bounds.right > result.bounds.right) {
+                result = displayFeature;
+              }
+            }
+            break;
+          }
+        }
+        break;
+        case Axis.vertical: {
+          switch (verticalDirection) {
+            case VerticalDirection.down: {
+              if (displayFeature.bounds.top < result.bounds.top) {
+                result = displayFeature;
+              }
+            }
+            break;
+            case VerticalDirection.up: {
+              if (displayFeature.bounds.bottom > result.bounds.bottom) {
+                result = displayFeature;
+              }
+            }
+            break;
+          }
+        }
+        break;
+      }
+    }
+    return result;
+  }
+
+  /// Returns only the display features that separate the screen into
+  /// sub-screens.
+  ///
+  /// A [DisplayFeature] separates the screen into sub-screens when both these
```

Comment:
This seems to leave out foldable devices with a continuous screen, but I may be misunderstanding

Reply:
If the screen is continuous the two panes don't need to split on the folding feature, so they are free to take the proportions provided to the widget, the same way they would on a tablet or larger screen.